### PR TITLE
Move to java 8

### PR DESCRIPTION
**Pull request by robert3005**
cc @uschi2000

### CODE REVIEW

File: junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java [hunk 1]
```diff
@@ -16,39 +16,31 @@
...
         for (int i = 0; i < numAttempts; ++i) {
-            lastException = scheduler.schedule(isReadyCallable, intervalMillis, TimeUnit.MILLISECONDS).get();
+            lastException = scheduler.schedule(target::isReady, intervalMillis, TimeUnit.MILLISECONDS).get();
             if (!lastException.isPresent()) {
                 return;
             }
         }
 
-        throw lastException.or(new IllegalStateException("Internal error (numAttempts == 0?)"));
+        lastException.orElseThrow(() -> new IllegalStateException("Internal error (numAttempts == 0?)"));
```
Comment (uschi2000) [inline junit-resource-poller/src/main/java/com/palantir/junit/ResourcePoller.java]: need to throw the thing.

**Review by uschi2000 — COMMENTED** [review]
one bug (throw thingy), otherwise looks good.

**Comment by robert3005** [comment]
Yup, updated at the same time you left the comment. Should be fixed now

**Comment by uschi2000** [comment]
Cool,  feel free to merge

**Comment by robert3005** [comment]
I don't have permissions to merge :(

**Comment by ash211** [comment]
@uschi2000 can you merge?

**Review by uschi2000 — APPROVED** [review]

**Pull request merged**